### PR TITLE
assorted server improvements

### PR DESCRIPTION
**Pull request by natenichols**
save error on failure

execute sends on websocket executor

### CODE REVIEW

File: src/webserver/Listener.h [hunk 1]
```diff
@@ -55,6 +55,16 @@ class Detector
     {
     }
 
+    inline void
+    fail(boost::system::error_code ec, char const* message)
+    {
+        if (ec == net::ssl::error::stream_truncated)
+            return;
+
+        std::cerr << "Detector failed: " << message << ec.message()
```
Comment (cjcobb23) [inline src/webserver/Listener.h]: Change this to log, not cerr

File: src/webserver/SslWsSession.h [hunk 2]
```diff
@@ -146,6 +146,8 @@ class SslWsUpgrader : public std::enable_shared_from_this<SslWsUpgrader>
         boost::beast::get_lowest_layer(https_).expires_after(
             std::chrono::seconds(30));
 
+        auto self = shared_from_this();
```
Comment (cjcobb23) [inline src/webserver/SslWsSession.h]: Is this used?

File: src/webserver/WsBase.h [hunk 3]
```diff
@@ -91,8 +85,21 @@ class WsSession : public WsBase,
...
+    bool sending_ = false;
     std::queue<std::string> messages_;
 
+    void
+    wsFail(boost::beast::error_code ec, char const* what)
+    {
+        if (!ec_ && ec != boost::asio::error::operation_aborted)
+        {
+            ec_ = ec;
+            std::cerr << "wsFail: " << what << ": " << ec.message();
```
Comment (cjcobb23) [inline src/webserver/WsBase.h]: Change to log

**Review by cjcobb23 — CHANGES_REQUESTED** [review]
So we no longer need locking because everything is executing on the same strand now?

**Comment by natenichols** [comment]
> So we no longer need locking because everything is executing on the same strand now?

That is correct.

**Review by cjcobb23 — APPROVED** [review]

**Pull request merged**